packages/amplify-cli-core/src/state-manager/stateManager.ts

```diff
@@ -27,6 +28,11 @@ export class StateManager {
 
   currentMetaFileExists = (projectPath?: string): boolean => fs.existsSync(pathManager.getCurrentAmplifyMetaFilePath(projectPath));
 
+  setDeploymentSecrets = (deploymentSecrets: $TSAny): void => {
+    const path = pathManager.getDeploymentSecrets();
+    JSONUtilities.writeJson(path, deploymentSecrets, { mode: 0o600 }); //set deployment secret file permissions to -rw-------
```

Comment:
can we store this permission level as a constant. I think we re-use this level of permissions during `amplify configure` as well